File: apps/web/src/core/domain/repositories/todo_repository.ts
import type { TodoEntity } from "../entities/todo_entity";

export interface TodoRepository {
  findAll(): Promise<TodoEntity[]>;
  findById(id: string): Promise<TodoEntity | null>;
  create(todo: TodoEntity): Promise<TodoEntity>;
  update(todo: TodoEntity): Promise<TodoEntity>;
  delete(id: string): Promise<void>;
}

File: apps/mobile/android/app/src/main/java/com/erluxman/productive/core/domain/repositories/TodoRepository.kt
package com.erluxman.productive.core.domain.repositories

import com.erluxman.productive.core.domain.entities.TodoEntity

interface TodoRepository {
    suspend fun getAll(): Result<List<TodoEntity>>
    suspend fun getById(id: String): Result<TodoEntity>
    suspend fun create(title: String): Result<TodoEntity>
    suspend fun update(todo: TodoEntity): Result<TodoEntity>
    suspend fun delete(id: String): Result<Unit>
}

